sunxi: power: enable DLDO1 switch as well

Icenowy opened the pull request:
(Note: two new commits added)

Enable DLDO1 for HDMI, enable PL2/4/6 for Wi-Fi, and rasie the delay time to successfully setup clock for A64.

Icenowy committed:
sunxi: power: enable DLDO1 switch as well

The DLDO1 switch enables power to the HDMI part of the SoC.
Enable this power rail unconditionally for now until we have proper
regulator support.

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>

Icenowy committed:
sunxi: enable PL2/4/6 on A64 boards

As all A64 development boards available now has Wi-Fi/BT power
controlled at PL2/4/6, enable them when it's A64.

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>

Icenowy committed:
sunxi: clocks: raise delay time

The original delay time is not enough, and easily hangs the system.

Add a lot time to it, so the clock can switch successfully.

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>

Icenowy committed:
sunxi: re-enable PLL_CPUX setup on H5

As the delay is extended, PLL_CPUX setup can now reliably work on H5.

Re-enable it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>

apritzel commented:
Hi Icenowy,

thanks for the patches, interesting find with the delays. I actually thought that a delay of "1" (=2000 cycles) should be plenty, since the original code and the manual speak of much less. Did you find this by experimentation or via some documentation/BSP code?

I have some reworks in the pipe (to support the I2C PMIC on the H5), so will try to stuff your patches into somehow.

Icenowy commented:
2017年1月30日 10:29于 Andre Przywara <notifications@github.com>写道：
>
> Hi Icenowy,
>
> thanks for the patches, interesting find with the delays. I actually thought that a delay of "1" (=2000 cycles) should be plenty, since the original code and the manual speak of much less. Did you find this by experimentation or via some documentation/BSP code?

By experimentation...

Firstly I added debugging print there, then the code mysteriously worked; then I deleted the print code and added the delay.

Without the delay change, even on A64 it cannot work here.

>
> I have some reworks in the pipe (to support the I2C PMIC on the H5), so will try to stuff your patches into somehow.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

apritzel commented:
I merged the HDMI power patch. The H5 fix is no longer needed thanks to Philipp's clock setup rework. For the PL GPIO setup: I am not fully convinced of it, I might merge it later together with some generic GPIO support, if that's still needed.
